apps/omg_watcher_info/lib/omg_watcher_info/db/block.ex

```diff
@@ -54,6 +58,21 @@ defmodule OMG.WatcherInfo.DB.Block do
     DB.Repo.aggregate(__MODULE__, :max, :blknum)
   end
 
+  @doc """
+    Gets a block specified by a hash.
+  """
+  def get(hash) do
+    query =
+      from(
+        __MODULE__,
+        where: [hash: ^hash],
+        preload: [
+          transactions: ^from(tx in DB.Transaction, order_by: :blknum),
```

Comment:
This is probably fine to preload transactions. But just wondering why you thought to do this or which design mock needs this? I see we need txn sum for a block, but not the actual txns in our design.

Just curious. If we don't have a solid use case, might be best to not preload at first.

Reply:
Ah, my bad....I'm mixing the `block.get` `block.all` endpoints up. 😄  This is all good.

On another note though, not sure how many txns per block we expect on average when we're running at full speed, but do we need to paginate txns for each block?

Reply:
@unnawut this starts to get messy with pagination though, no? In the UI design this won't usually be optional and we'll want to paginate.

Reply:
Can you give some code example of what you mean? (or the solution you have in mind)

Reply:
Maybe your code example already answers my question`query = Transaction |> Repo.get_by(map)` where map will have `limit` and `page` variables. Is that correct? If so, roger that and nvm.

If not, what I was thinking is, the BE use case for `block.get` will always want txns preloaded so the txn list can be displayed in the UI. This list will need to be paginated, similar to how `blocks.all`, `txns.all` and `deposits.all` will be.

(At some point filtering transactions in a block on certain properties may also be useful...but that's getting a bit into the future...)

Reply:
Based on the fact that we can have a crazy huge amount of transactions per block, some kind of pagination needs to be there from the beginning. Two options come to my mind:

1. Keep the `/block.get` as it is now (with the block containing a list of transactions) but paginate those transactions and allow a user to specify those pagination params.
2. Forget about returning transactions with `/block.get` and instead simply return the blocks details. The list of transactions will be retrievable with the existing `/transactions.all` endpoint, which should already support pagination, and would simply need the addition of a filter by `blk_num` or block hash to match the frontend needs.

Reply:
Thank you for comments all. The `/transaction.all` endpoint already supports `blknum` as a  a filter. See [here](https://developer.omisego.co/elixir-omg/docs-ui/?url=master%2Foperator_api_specs.yaml&urls.primaryName=master%2Finformational_api_specs#/Transaction/transactions_all). 

Discussed with @unnawut and @jrhite and we think the best course of action is @T-Dnzt's option 2: the client will call `/block.get` to get a block's basic info and then `transaction.all` with a `blknum` param.

Reply:
Thanks for everyone's input and @T-Dnzt for the suggestion! 

As we won't be needing any transaction pre-loading, I'll delete those parts of the code and close some of the related comments below.